### PR TITLE
Converting response to raise a ProblemException

### PR DESCRIPTION
**Pull request by ConstantinoSchillebeeckxSimpleRose**
Per https://github.com/zalando/connexion/pull/955#issuecomment-542302521, I've rebased a stale [PR](https://github.com/zalando/connexion/pull/955) and made the proper fixes.

Some tests are failing on Python >= 3.5:

```
=================================== FAILURES ===================================
________________________ test_aiohttp_problems[pyloop] _________________________
aiohttp_app = <connexion.apps.aiohttp_app.AioHttpApp object at 0x7fe2ef83a828>
aiohttp_client = <function aiohttp_client.<locals>.go at 0x7fe2eee20400>
    @asyncio.coroutine
    def test_aiohttp_problems(aiohttp_app, aiohttp_client):
        # TODO: This is a based on test_errors.test_errors(). That should be refactored
        #       so that it is parameterized for all web frameworks.
        app_client = yield from aiohttp_client(aiohttp_app.app)  # type: aiohttp.test_utils.TestClient
    
        greeting404 = yield from app_client.get('/v1.0/greeting')  # type: aiohttp.ClientResponse
>       assert greeting404.content_type == 'application/problem+json'
E       AssertionError: assert 'text/plain' == 'application/problem+json'
E         - text/plain
E         + application/problem+json
tests/aiohttp/test_aiohttp_errors.py:31: AssertionError
```
I'm not familiar enough with the code base to quickly fix this, so I may look at it later unless someone can suggest a quick fix.

### CODE REVIEW

File: tests/api/test_responses.py [hunk 1]
```diff
@@ -156,19 +156,6 @@ def test_redirect_response_endpoint(simple_app):
...
-    assert resp.status_code == 200
-    response = json.loads(resp.data.decode('utf-8', 'replace'))
-    assert response['stack'] == {'image_version': 'default_image'}
-
-    resp = app_client.post('/v1.0/test-default-integer-body')
-    assert resp.status_code == 200
-    response = json.loads(resp.data.decode('utf-8', 'replace'))
-    assert response == 1
-
-
```
Comment (hjacobs) [inline tests/api/test_responses.py]: why is this removed?

Comment (cognifloyd) [reply tests/api/test_responses.py]: Here's what @badcure had to say about why he removed it:
https://github.com/zalando/connexion/pull/955#issuecomment-495367520
Removed in https://github.com/zalando/connexion/pull/955/commits/cdcb0fda36d1a8dc0a75e47934a67a0c50f4ad6d (part of original PR #955) or https://github.com/zalando/connexion/pull/1054/commits/098854d9276276c225727885bde64335eecfc2fb (in this PR #1054)

File: connexion/apis/aiohttp_api.py [hunk 2]
```diff
@@ -51,36 +51,22 @@ def _generic_problem(http_status: HTTPStatus, exc: Exception = None):
...
+            body=json.dumps(oauth_error.description).encode(),
+            content_type='application/problem+json'
+        )
+    except ProblemException as prob_error:
+        problem_response = problem_from_object(prob_error)
+        return web.Response(
+            status=problem_response.status_code,
+            body=json.dumps(problem_response.body).encode(),
+            content_type=problem_response.content_type
+        )
```
Comment (cognifloyd) [inline connexion/apis/aiohttp_api.py]: This was merged wrong. Please make sure that all of the cases handled in problems_middleware are also handled here and also update _generic_problem above.

**Comment by ConstantinoSchillebeeckxSimpleRose** [comment]
Closing for favor of https://github.com/zalando/connexion/pull/955
